src/main.ts

```diff
@@ -64,3 +61,29 @@ function isGhes(): boolean {
   );
   return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
 }
+
+function parseNodeVersion(): string {
+  let nodeVersion = core.getInput('node-version') || core.getInput('version');
+
+  if (!nodeVersion) {
+    if (fs.existsSync('.nvmrc')) {
+      // Read from .nvmrc
+      nodeVersion = fs.readFileSync('.nvmrc', 'utf8').trim();
+      console.log(`Using ${nodeVersion} as input from file .nvmrc`);
+    } else if (fs.existsSync('.tool-versions')) {
+      // Read from .tool-versions
+      const toolVersions = fs.readFileSync('.tool-versions', 'utf8').trim();
+      const nodeLine = toolVersions
+        .split(/\r?\n/)
+        .filter(e => e.match(/^nodejs\s/))[0];
+      nodeVersion = nodeLine.match(/^nodejs\s+(.+)$/)![1];
```

Comment:
`.match()` returns `RegExpMatchArray | null`, so rather than using definite assignment assertion (`!`), perhaps use the optional chain (`?`) to query the RegExpMatchArray. If nothing is returned here, it will assign `undefined` to `nodeVersion`. 

This does then throw into question the flow of these statements, because if I have a `.tool-versions` with the `nodejs` key in (I think that's what you're testing?) but no value, then it will return undefined.

I would revisit this by allowing TS to guide you with the types & then making sure that a string is definitely being returned.

```suggestion
      nodeVersion = nodeLine.match(/^nodejs\s+(.+)$/)?.[1];
```